File: backend/src/main/java/dev/diegobarrioh/forma/application/PlanAcceptance.java
package dev.diegobarrioh.forma.application;

/**
 * What the first screen after login needs to know: whether there is a plan waiting to be started,
 * and what it is called.
 *
 * @param pending whether a plan is written but not yet switched on
 * @param planName the name of that plan, or {@code null} when there is nothing to offer
 */
public record PlanAcceptance(boolean pending, String planName) {

  private static final PlanAcceptance NOTHING = new PlanAcceptance(false, null);

  /** Nothing to offer: no plan, or one the user already accepted. */
  public static PlanAcceptance nothing() {
    return NOTHING;
  }

  /** A plan is waiting for an answer. */
  public static PlanAcceptance of(String planName) {
    return new PlanAcceptance(true, planName);
  }
}

File: frontend/src/api/planAcceptance.ts
/**
 * Plan acceptance (V58): whether a plan is written and waiting to be started, and starting it.
 *
 * <p>Built on the shared {@link apiClient} boundary (ADR-006 — no ad-hoc `fetch`).
 */
import { apiClient, type ApiClient } from './client';

/**
 * What the app is offered on first login.
 *
 * <p>`planName` is absent when there is nothing to offer — either the account already accepted, or
 * it has no plan at all. The second case is not an error: it is an account whose plan generation
 * produced nothing, and the screens say so rather than offering a plan that is not there.
 */
export interface PlanAcceptance {
  readonly pending: boolean;
  readonly planName?: string;
}

const PATH = '/api/v1/plan-acceptance';

/** Whether a plan is waiting to be started. Never 404s. */
export function getPlanAcceptance(client: ApiClient = apiClient): Promise<PlanAcceptance> {
  return client.request<PlanAcceptance>(PATH);
}

/** Starts the plan: the server activates it and records the acceptance in one go. */
export function acceptPlan(client: ApiClient = apiClient): Promise<void> {
  return client.request<void>(PATH, { method: 'POST' });
}
